src/App.tsx

```diff
@@ -1,223 +1,261 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useCallback,
+  useMemo,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
 
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoItem } from './components/TodoItem/TodoItem';
+import { Footer } from './components/Footer/Footer';
+import { ErrorMessage } from './components/Error/Error';
+import {
+  getTodos,
+  addTodo,
+  deleteTodo,
+  editTodo,
+} from './api/todos';
+import { FilterType } from './types/Filter';
+import { Todo } from './types/Todo';
+import { Error } from './types/Error';
+
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [selectedTodoId, setSelectedTodoId] = useState<number[]>([]);
+
+  const activeTodosAmount = useMemo(() => (
+    todos.filter(todo => !todo.completed).length
+  ), [todos]);
+
+  const isTodoCompleted = useMemo(() => (
+    todos.some(todo => todo.completed)
+  ), [todos]);
+
+  const isEveryTodoCompleted = useMemo(() => (
+    todos.every(todo => todo.completed)
+  ), [todos]);
+
+  const visibleTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterType) {
+        case FilterType.Active:
+          return !todo.completed;
+
+        case FilterType.Completed:
+          return todo.completed;
+
+        default:
+          return todo;
+      }
```

Comment:
Don't start filtering if FilterType.All or default case
```suggestion
  switch (filterType) {
    case FilterType.Active:
      return todos.filter(todo => (
        !todo.completed;
      )
      
      .....

    case FilterType.ALL:
    default:
      return todos;
    }
```